react/features/calendar-sync/components/MeetingList.native.js

```diff
@@ -0,0 +1,237 @@
+
+    /**
+     * Implements React Component's componentWillReceiveProps function.
+     *
+     * @inheritdoc
+     */
+    componentWillReceiveProps(newProps) {
+        // This is a conditional logic to refresh the calendar entries (thus
+        // to request access to calendar) on component first receives a
+        // displayed=true prop - to avoid requesting calendar access on
+        // app start.
+        if (!this._initialLoaded
+                && newProps.displayed
+                && !this.props.displayed) {
+            const { dispatch } = this.props;
+
+            this._initialLoaded = true;
+            dispatch(refreshCalendarEntryList());
+        }
+    }
+
+    /**
+     * Implements the React Components's render method.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const { disabled } = this.props;
+
+        return (
+            <NavigateSectionList
+                disabled = { disabled }
+                onPress = { this._onPress }
+                onRefresh = { this._onRefresh }
+                sections = { this._toDisplayableList() } />
+        );
+    }
+
+    _onPress: string => Function
+
+    /**
+     * Handles the list's navigate action.
+     *
+     * @private
+     * @param {string} url - The url string to navigate to.
+     * @returns {void}
+     */
+    _onPress(url) {
+        const { dispatch } = this.props;
+
+        dispatch(appNavigate(url));
+    }
+
+    _onRefresh: () => void
+
+    /**
+     * Callback to execute when the list is doing a pull-to-refresh.
+     *
+     * @private
+     * @returns {void}
+     */
+    _onRefresh() {
+        const { dispatch } = this.props;
+
+        dispatch(refreshCalendarEntryList());
+    }
+
+    _toDisplayableItem: Object => Object
+
+    /**
+     * Creates a displayable object from an event.
+     *
+     * @private
+     * @param {Object} event - The calendar event.
+     * @returns {Object}
+     */
+    _toDisplayableItem(event) {
+        return {
+            key: `${event.id}-${event.startDate}`,
+            lines: [
+                event.url,
+                this._toDateString(event)
+            ],
+            title: event.title,
+            url: event.url
+        };
+    }
+
+    _toDisplayableList: () => Array<Object>
+
+    /**
+     * Transforms the event list to a displayable list
+     * with sections.
+     *
+     * @private
+     * @returns {Array<Object>}
+     */
+    _toDisplayableList() {
+        const { _eventList, t } = this.props;
+        const now = Date.now();
+        const nowSection = NavigateSectionList.createSection(
+            t('calendarSync.now'),
+            'now'
+        );
+        const nextSection = NavigateSectionList.createSection(
+            t('calendarSync.next'),
+            'next'
+        );
+        const laterSection = NavigateSectionList.createSection(
+            t('calendarSync.later'),
+            'later'
+        );
+
+        if (_eventList && _eventList.length) {
```

Comment:
I've generally preferred the defaultProps approach so that I know something is defined and don't have to worry about additional truthy checks. It lets me be....more reckless than I should be!

Reply:
Yeah, good point, I change it, thanks!